src/jsTyping/jsTyping.ts

```diff
@@ -149,8 +149,10 @@ namespace ts.JsTyping {
             const nodeModulesPath = combinePaths(searchDir, "node_modules");
             getTypingNamesFromPackagesFolder(nodeModulesPath, filesToWatch);
         });
-        getTypingNamesFromSourceFileNames(fileNames);
-
+        // filename-based ATA must be explicitly disabled
+        if(typeAcquisition.inferTypings !== false) {

```

Comment:
In general i am confused on how this is suppose to work.. 
If this is suppose to work just like compilerOptions parallel then we have compilerOptions per projectRoot setting and you would want something similar.

Reply:
The intention here was to make the editor settings global rather than project-specific. If users want per project configuration I would expect them to use a tsconfig.